### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by pleshevskiy**
- renamed `domain/erase` to `domain/overwrite`. I'll add a canonical erase later, which will use overwrite itself.
- changed all `match` patterns that are used as `mar_err`
- added copy_from_slice instead of a manual copy loop

### CODE REVIEW

File: dexios/src/subcommands/decrypt.rs [hunk 1]
```diff
@@ -206,10 +206,8 @@ pub fn stream_mode(input: &str, output: &str, params: &CryptoParams) -> Result<(
             })?;
 
             let mut master_key = [0u8; 32];
-
-            for (i, byte) in master_key_decrypted.iter().enumerate() {
-                master_key[i] = *byte;
-            }
+            let len = 32.min(master_key_decrypted.len());
```
Comment (brxken128) [inline dexios/src/subcommands/decrypt.rs:209]: I'm not too sure what this line does, please could you explain it?

Why would 32 not work on it's own? Thank you!

Comment (pleshevskiy) [reply dexios/src/subcommands/decrypt.rs:209]: I'm reinsuring) a panic in the CLI would not be particularly pleasant...
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=c101f4fd89ef4e5a25618a7038dc4244

Comment (brxken128) [reply dexios/src/subcommands/decrypt.rs:209]: Makes sense! Thank you :)

**Comment by brxken128** [comment]
All looks great my friend, nice work as usual!

**Comment by brxken128** [comment]
Feel free to add yourself to the authors in Cargo.toml if you'd like! You've done massive amounts of work friend :)

**Comment by pleshevskiy** [comment]
Thank you) but I don't think **1000 LOC** is "massive amount". First of all, I have to extract most of the logic from the cli :)

**Pull request merged**